src/features/tokens/SelectTokenIdField.tsx

```diff
@@ -0,0 +1,109 @@
+import Image from 'next/image';
+import { useState } from 'react';
+import { useAccount } from 'wagmi';
+
+import { Spinner } from '../../components/animation/Spinner';
+import { Modal } from '../../components/layout/Modal';
+import ChevronIcon from '../../images/icons/chevron-down.svg';
+
+import { useTokenIdBalance } from './useTokenBalance';
+
+type Props = {
+  name: string;
+  chainId: ChainId;
+  tokenAddress: Address;
+  disabled?: boolean;
+};
+
+export function SelectTokenIdField({ name, chainId, tokenAddress, disabled }: Props) {
+  const [, , helpers] = useField<number>(name);
+  const [tokenId, setTokenId] = useState<string | undefined>(undefined);
+  const { address } = useAccount();
+  const handleChange = (newTokenId: string) => {
+    helpers.setValue(parseInt(newTokenId));
+    setTokenId(newTokenId);
+  };
+
+  const { isLoading, tokenIds } = useTokenIdBalance(chainId, tokenAddress, address);
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const onClick = () => {
+    if (!disabled) setIsModalOpen(true);
+  };
+
+  return (
+    <div className="flex flex-col items-center">
+      <button type="button" className={styles.base} onClick={onClick}>
+        <div className="flex items-center">
+          <span className={`ml-2 ${!tokenId && 'text-slate-400'}`}>
+            {tokenId ? tokenId : 'Select Token Id'}
+          </span>
+        </div>
+        <Image src={ChevronIcon} width={12} height={8} alt="" />
+      </button>
+      <SelectTokenIdModal
+        isOpen={isModalOpen}
+        tokenIds={tokenIds}
+        isLoading={isLoading}
+        close={() => setIsModalOpen(false)}
+        onSelect={handleChange}
+      />
+    </div>
+  );
+}
+
+export function SelectTokenIdModal({
+  isOpen,
+  tokenIds,
+  isLoading,
+  close,
+  onSelect,
+}: {
+  isOpen: boolean;
+  tokenIds: string[] | null | undefined;
+  isLoading: boolean;
+  close: () => void;
+  onSelect: (tokenId: string) => void;
+}) {
+  const onSelectTokenId = (tokenId: string) => {
+    return () => {
+      onSelect(tokenId);
+      close();
+    };
+  };
+
+  return (
+    <Modal isOpen={isOpen} title="Select Token Id" close={close}>
+      <div className="mt-2 flex flex-col space-y-1">
+        {isLoading ? (
+          <div className="my-24 flex flex-col items-center">
+            <Spinner />
+            <h3 className="mt-5 text-sm text-gray-500">Finding token IDs</h3>
+          </div>
+        ) : tokenIds && tokenIds.length !== 0 ? (
+          tokenIds.map((id) => (
+            <button
+              key={id}
+              className="py-1.5 px-2 text-sm flex items-center rounded hover:bg-gray-100 active:bg-gray-200 transition-all duration-200"
+              onClick={onSelectTokenId(id)}
+            >
+              <span className="ml-2">{id}</span>
+            </button>
+          ))
+        ) : (
+          <div className="py-1.5 px-2 text-sm text-gray-500 transition-all duration-200">
+            No tokenIds found
```

Comment:
Like the other comments, please use 'token id` and not 'tokenid' in the UI. Missed this one before but please check for other instances